management/server/account.go

```diff
@@ -658,6 +711,55 @@ func (am *DefaultAccountManager) UpdateAccountSettings(accountID, userID string,
 	return updatedAccount, nil
 }
 
+func (am *DefaultAccountManager) peerLoginExpirationJob(accountID string) func() (bool, time.Duration) {
+	return func() (bool, time.Duration) {
+		unlock := am.Store.AcquireAccountLock(accountID)
+		defer unlock()
+
+		account, err := am.Store.GetAccount(accountID)
+		if err != nil {
+			log.Errorf("failed getting account %s expiring peers", account.Id)
+			// todo return retry?
+			return false, 0
+		}
+
+		var peerIDs []string
+		for _, peer := range account.GetExpiredPeers() {
+			peerIDs = append(peerIDs, peer.ID)
+			peer.MarkLoginExpired(true)
+			account.UpdatePeer(peer)
+			err = am.Store.SavePeerStatus(account.Id, peer.ID, *peer.Status)
+			if err != nil {
+				log.Errorf("failed saving peer status while expiring peer %s", peer.ID)
+				// todo return retry?
+				return false, 0
+			}
+		}
```

Comment:
Not sure on this one but I think we do way to much computing overhead here.

`account.GetExpiredPeers()` will return the complete list of expired peers. Even peers that have been expired for months. So in case we we have lots of zombies in the account that are not used anymore but are doing database overrides and peer updates for all of them on any run with the same (unchanged) data.

I would add additional check for status changes at the top of the loop:
```
if peer.Status.LoginExpired {
     continue
}
```